sklearn/impute/_base.py

```diff
@@ -240,20 +240,28 @@ def fit(self, X, y=None):
                                  "== 0 and input is sparse. Provide a dense "
                                  "array instead.")
             else:
-                self.statistics_ = self._sparse_fit(X,
-                                                    self.strategy,
-                                                    self.missing_values,
-                                                    fill_value)
+                self.statistics_, missing_mask = self._sparse_fit(
+                    X, self.strategy, self.missing_values, fill_value)
+
+                if self.add_indicator:
+                    sparse_constructor = (sparse.csr_matrix
+                                          if X.format == 'csr'
+                                          else sparse.csc_matrix)
+                    missing_mask = sparse_constructor(
+                        (missing_mask, X.indices.copy(), X.indptr.copy()),
```

Comment:
Do we really need to copy?

Reply:
As in: isn't this about to be stacked to the imputed data?

Reply:
Yup, I don't think we need to copy it, I just can't remember why I did so. I'll remove it in the next commit.

Reply:
Now I remember, why I was copying it. If I don't copy, the transform does not stack the imputed data to `X`, but it does so when we copy the `indices` and `indptr`. A weird behaviour, I'm trying to figure it out.

Thanks.